src/Components/Markdown.re

```diff
@@ -0,0 +1,130 @@
+/* Markdown.re
+
+    This adapts Revery's built in Markdown component to our code highlighters
+   */
+
+open Revery.UI;
+open Revery.UI.Components;
+open Oni_Core;
+open Oni_Syntax;
+
+module Colors = Feature_Theme.Colors;
+
+open {
+       let syntaxHighlighter =
+           (
+             ~tokenTheme,
+             ~colorTheme,
+             ~languageInfo,
+             ~grammars,
+             ~language,
+             lines,
+           ) => {
+         let grammarRepository =
```

Comment:
Ideally, we shouldn't need to recreate the `grammarRepository` every time - this means we'll have to reload grammars every time we render. We have a common instance of `grammarRepository` right now - can we use that here?

Reply:
Oof that was my bad! I moved it to the state but didn't change it in the component :facepalm. Luckily it's an easy fix!

Reply:
Actually I'm realizing that here it's creating a Textmate grammar repository from an Oni_Syntax one. We carry around the Oni_Syntax one which gets passed here, should I also add a Textmate one to state? I can also make it stateful, so it doesnt re-create it every render.

Reply:
Ah, got it - so it's using our common grammar repository (taking advantage of the cache) - just wrapping it. That should be fine!